common/persistence/PersistenceManager.php

```diff
@@ -0,0 +1,131 @@
+        'phpfile'    => 'common_persistence_PhpFileDriver',
+        'SqlKvWrapper' => 'common_persistence_SqlKvDriver',
+        'no_storage' => 'common_persistence_InMemoryKvDriver',
+        'no_storage_adv' => 'common_persistence_InMemoryAdvKvDriver'
+    );
+
+    /**
+     * @var array
+     */
+    private $persistences = array();
+
+    /**
+     * Returns TRUE if the requested persistence exist, otherwise FALSE.
+     *
+     * @param string $persistenceId
+     * @return bool
+     */
+    public function hasPersistence($persistenceId)
+    {
+        $persistenceList = $this->getOption(static::OPTION_PERSISTENCES);
+        return isset($persistenceList[$persistenceId]);
+    }
+
+    /**
+     * Registers a new persistence.
+     *
+     * @param string $persistenceId
+     * @param array  $persistenceConf
+     */
+    public function registerPersistence($persistenceId, array $persistenceConf)
+    {
+        // wrap pdo drivers in dbal
+        if (strpos($persistenceConf['driver'], 'pdo_') === 0) {
+            $persistenceConf = [
+                'driver' => 'dbal',
+                'connection' => $persistenceConf,
+            ];
+        }
+
+        if (isset($persistenceConf['connection']['driver']) && $persistenceConf['connection']['driver'] === 'pdo_mysql') {
+            $persistenceConf['connection']['charset'] = 'utf8';
+        }
+
+        $configs = $this->getOption(self::OPTION_PERSISTENCES);
+        $configs[$persistenceId] = $persistenceConf;
+        $this->setOption(self::OPTION_PERSISTENCES, $configs);
+    }
+
+    /**
+     * @return \common_persistence_Persistence
+     */
+    public function getPersistenceById($persistenceId)
+    {
+        if(!isset($this->persistences[$persistenceId])) {
+            $this->persistences[$persistenceId] = $this->createPersistence($persistenceId);
+        }
+        return $this->persistences[$persistenceId];
+    }
+
+    /**
+     * @param string $persistenceId
+     * @throws \common_Exception
+     * @return \common_persistence_Persistence
+     */
+    private function createPersistence($persistenceId) {
+        $configs = $this->getOption(self::OPTION_PERSISTENCES);
+        if (isset($configs[$persistenceId])) {
+            $config = $configs[$persistenceId];
+            $driverStr = $config['driver'];
+            
+            $driverClassName = isset(self::$driverMap[$driverStr])
+                ? self::$driverMap[$driverStr]
+                : $driverStr;
+             
+            if (!class_exists($driverClassName)){
+                throw new \common_exception_Error('Driver '.$driverStr.' not found check your database configuration');
+            }
+            $driver = $this->propagate(new $driverClassName());
+            return $driver->connect($persistenceId, $config);
+        } else {
+            throw new \common_Exception('Persistence Configuration for persistence '.$persistenceId.' not found');
```

Comment:
Please use early exit instead of else:
```
if (!isset($configs[$persistenceId])) {
    throw new \common_Exception('Persistence Configuration for persistence '.$persistenceId.' not found');
}
```
and remove one level of cpomplexity.